src/resolvers/viewsInfo.ts

```diff
@@ -123,10 +144,12 @@ export class VideoViewsInfosResolver {
 }
 
 const mapMostViewedArray = (views: Record<string, number>, limit?: number) =>
-  Object.keys(views)
-    .map((id) => ({ id, views: views[id] }))
-    .sort((a, b) => (a.views > b.views ? -1 : 1))
-    .slice(0, limit)
+  views
+    ? Object.keys(views)
+        .map((id) => ({ id, views: views[id] }))
+        .sort((a, b) => (a.views > b.views ? -1 : 1))
+        .slice(0, limit)
+    : []
 
 const filterAllViewsByPeriod = (ctx: OrionContext, period?: number): Partial<UnsequencedVideoEvent>[] => {
```

Comment:
This was mostly worked on in the previous PR, but we will need to improve upon this logic. Assume Orion hold 1M events from the past year. Assume only 100 of them are from yesterday. With the current implementation, every time you want to get events from last day, you will end up iterating over all of them. We now that `getAllViewsEvents` will return a sorted list. I think we can safely iterate backwards and break out of the loop on first event that is outside of the specified period, right?